src/api/Interpret.cc

```diff
@@ -899,21 +899,23 @@ bool Interpret::defineFun(const ASTNode& n)
     const char* fname = name_node.getValue();
 
     // Get the argument sorts
-    vec<SRef> arg_sorts;
-    vec<PTRef> arg_trs;
+    struct DefineFunArg {
+        PTRef inner;
+        std::string formalName;
+    };
+    std::vector<DefineFunArg> args;
     for (auto childNodePtr : *args_node.children) {
         ASTNode const & childNode = *childNodePtr;
         assert(childNode.children->size() == 1);
-        std::string varName = childNode.getValue();
         ASTNode const & sortNode = **(childNode.children->begin());
         SRef sortRef = sortFromASTNode(sortNode);
         if (sortRef == SRef_Undef) {
             notify_formatted(true, "Undefined sort %s in function %s", sortSymbolFromASTNode(sortNode).name.c_str(), fname);
             return false;
         }
-        arg_sorts.push(sortRef);
-        PTRef pvar = logic->mkVar(arg_sorts.last(), varName.c_str());
-        arg_trs.push(pvar);
+        PTRef pvar = logic->mkVar(sortRef, TemplateFunction::nextFreeArgumentName().c_str());
+        std::string formalName = childNode.getValue();
```

Comment:
It doesn't seem to change much, but why do we use string instead of SRef now?
Was it just unoptimal before?